### PR TITLE
Add a progress bar for the total download of shards

### LINKED ISSUE #22047: add progress bar to the sharded model download status
### Feature request

`from_pretrained('bigscience/bloom')` is taking forever the first time until it's cached (~350GB) - I thought that perhaps with 72 shards it'd be awesome to have an overall progress bar (in addition to the each shard download progress bar) to know where things stand and how many hours the coffee break should last.

Thank you!

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR adds a feature requested in #22047 and fixes a small bug I encountered while testing it.

**The feature**: a new progress bar is added when loading a sharded checkpoint that gives the overall progress.

**The bug**: when passing along `force_download=True`, the files were not downloaded if cached, because an early return in `cached_file` returned the cached file.

Fixes #22047

### CODE REVIEW

**Comment by stas00** [comment]
I'm not observing any changes with this branch:

I tried:
```
python -c 'import sys; from transformers import AutoModel; AutoModel.from_pretrained(sys.argv[1], revision="sharded")' t5-11b
```

after deleting the cache.

**Comment by sgugger** [comment]
Woops, wrong check for the file being cached.

**Comment by stas00** [comment]
It works great! Thank you, Sylvain!

As you can see it moves the outer progress bar with the inner bar, so it doesn't matter how many shards there are.
snapshot_50

snapshot_51

**Review by stas00 — APPROVED** [review]
Looks awesome!

File: src/transformers/utils/hub.py [hunk 1]
```diff
@@ -390,7 +390,7 @@ def cached_file(
     if isinstance(cache_dir, Path):
         cache_dir = str(cache_dir)
 
-    if _commit_hash is not None:
+    if _commit_hash is not None and not force_download:
```
Comment (LysandreJik) [inline src/transformers/utils/hub.py:393]: Nice catch :)

**Review by LysandreJik — APPROVED** [review]
LGTM! Yay for intuitive progress bars!

**Pull request merged**